<launch>

  <include file="$(find mrs_simulation)/launch/simulation.launch">
    <arg name="gui" value="true" />
  </include>

  <arg name="SENSORS" default="garmin_down"/>
  <arg name="WORLD_NAME" default="simulation" />
  <arg name="RUN_TYPE" default="simulation" />
  <arg name="ODOMETRY_TYPE" default="gps" />
  <arg name="UAV_TYPE" default="t650" />
  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav)"/>
  <arg name="custom_config" default="" />
  <arg name="debug" default="false" />
  <arg name="start_with_takeoff" default="true" />

  <include file="$(find mrs_uav_general)/launch/core.launch">

    <arg name="SENSORS" default="$(arg SENSORS)" />
    <arg name="WORLD_NAME" default="$(arg WORLD_NAME)" />

    <arg name="config_mpc_tracker" default="$(find mrs_uav_testing)/test/custom_configs/mpc_tracker.yaml" />
    <arg name="config_odometry" default="$(find mrs_uav_testing)/test/custom_configs/odometry.yaml" />

  </include>

  <group ns="$(arg UAV_NAME)">

    <test pkg="mrs_uav_testing" type="ControlTestRostest" test-name="control_test" time-limit="600.0">

      <rosparam file="$(find mrs_uav_testing)/config/control_test.yaml" />
      <rosparam file="$(find mrs_uav_testing)/config/control_test_rostest.yaml" />

        <!-- Load a user param file -->
      <rosparam if="$(eval not arg('custom_config') == '')" file="$(arg custom_config)" />

      <param name="start_with_takeoff" value="$(arg start_with_takeoff)" />

        <!-- subscribers -->
      <remap from="~odometry_in" to="odometry/odom_main" />
      <remap from="~position_command_in" to="control_manager/position_cmd" />
      <remap from="~control_manager_diagnostics_in" to="control_manager/diagnostics" />
      <remap from="~spawner_diagnostics_in" to="/mrs_drone_spawner/diagnostics" />

        <!-- services out -->
      <remap from="~goto_out" to="control_manager/goto" />
      <remap from="~goto_relative_out" to="control_manager/goto_relative" />
      <remap from="~goto_altitude_out" to="control_manager/goto_altitude" />
      <remap from="~set_heading_out" to="control_manager/set_heading" />
      <remap from="~set_heading_relative_out" to="control_manager/set_heading_relative" />
      <remap from="~switch_tracker_out" to="control_manager/switch_tracker" />
      <remap from="~motors_out" to="control_manager/motors" />
      <remap from="~arm_out" to="mavros/cmd/arming" />
      <remap from="~offboard_out" to="mavros/set_mode" />
      <remap from="~takeoff_out" to="uav_manager/takeoff" />
      <remap from="~land_out" to="uav_manager/land" />
      <remap from="~land_home_out" to="uav_manager/land_home" />
      <remap from="~goto_trajectory_start_out" to="control_manager/goto_trajectory_start" />
      <remap from="~start_trajectory_tracking_out" to="control_manager/start_trajectory_tracking" />
      <remap from="~stop_trajectory_tracking_out" to="control_manager/stop_trajectory_tracking" />
      <remap from="~resume_trajectory_tracking_out" to="control_manager/resume_trajectory_tracking" />
      <remap from="~switch_estimator_out" to="odometry/change_odometry_source" />
      <remap from="~spawn_out" to="/mrs_drone_spawner/spawn" />

        <!-- services in -->
      <remap from="~start_in" to="automatic_start/start" />

        <!-- services and topics combined -->
      <remap from="~trajectory_reference_out" to="control_manager/trajectory_reference" />
      <remap from="~reference_out" to="control_manager/reference" />

    </test>

  </group>

</launch>
